SpiceSharp/Components/RLC/RES/TemperatureBehavior.cs

```diff
@@ -38,47 +71,48 @@ public override void Setup(Simulation simulation, SetupDataProvider provider)
+            ModelParameters = mbp;
         }
         
         /// <summary>
         /// Execute behavior
         /// </summary>
         /// <param name="simulation">Base simulation</param>
-        public override void Temperature(BaseSimulation simulation)
+        public void Temperature(BaseSimulation simulation)
         {
 			if (simulation == null)
 				throw new ArgumentNullException(nameof(simulation));
 
             double factor;
-            double resistance = _bp.Resistance;
+            double resistance = BaseParameters.Resistance;
 
             // Default Value Processing for Resistor Instance
-            if (!_bp.Temperature.Given)
-                _bp.Temperature.RawValue = simulation.RealState.Temperature;
-            if (!_bp.Width.Given)
-                _bp.Width.RawValue = _mbp?.DefaultWidth ?? 0.0;
+            if (!BaseParameters.Temperature.Given)
+                BaseParameters.Temperature.RawValue = simulation.RealState.Temperature;
+            if (!BaseParameters.Width.Given)
+                BaseParameters.Width.RawValue = ModelParameters?.DefaultWidth ?? 0.0;
 
-            if (_mbp != null)
+            if (ModelParameters != null)
             {
-                if (!_bp.Resistance.Given)
+                if (!BaseParameters.Resistance.Given)
                 {
-                    if (_mbp.SheetResistance.Given && _mbp.SheetResistance > 0 && _bp.Length > 0)
-                        resistance = _mbp.SheetResistance * (_bp.Length - _mbp.Narrow) / (_bp.Width - _mbp.Narrow);
+                    if (ModelParameters.SheetResistance.Given && ModelParameters.SheetResistance > 0 && BaseParameters.Length > 0)
+                        resistance = ModelParameters.SheetResistance * (BaseParameters.Length - ModelParameters.Narrow) / (BaseParameters.Width - ModelParameters.Narrow);
                     else
                     {
                         CircuitWarning.Warning(this, "{0}: resistance=0, set to 1000".FormatString(Name));
                         resistance = 1000;
                     }
                 }
 
-                var difference = _bp.Temperature - _mbp.NominalTemperature;
+                var difference = BaseParameters.Temperature - ModelParameters.NominalTemperature;
 
-                if (_mbp.ExponentialCoefficient.Given)
-                    factor = Math.Pow(1.01, _mbp.ExponentialCoefficient * difference);
+                if (ModelParameters.ExponentialCoefficient.Given)
+                    factor = Math.Pow(1.01, ModelParameters.ExponentialCoefficient * difference);
                 else
-                    factor = 1.0 + _mbp.TemperatureCoefficient1 * difference + _mbp.TemperatureCoefficient2 * difference * difference;
+                    factor = 1.0 + ModelParameters.TemperatureCoefficient1 * difference + ModelParameters.TemperatureCoefficient2 * difference * difference;
             }
             else
             {
```

Comment:
@svenboulanger Below there is:   if (resistance < 1e-12) resistance = 1e-12; 
Shouldn't we use Gmin parameter from BaseConfiguration?

Reply:
We can't, because Gmin from the BaseConfiguration can in some cases be set to 0. Perhaps we could use `Math.Min(BaseConfiguration.Gmin, 1e-14)` or something? But it wouldn't make much difference...

Reply:
@svenboulanger  OK, but maybe introduce a const for `1e-12`?